crates/server/src/authority/catalog.rs

```diff
@@ -485,8 +505,14 @@ async fn build_response(
     let mut response_header = Header::response_from_request(request_header);
     response_header.set_authoritative(authority.zone_type().is_authoritative());
 
-    debug!("performing {} on {}", query, authority.origin());
-    let future = authority.search(request_info, lookup_options);
+    let future = authority.search(request_info, lookup_options).await;
```

Comment:
I added a comment to call this change out, but, I don't see a way to keep the execution deferred in build_response - since it triggers a call to send either an authoritative or forwarded response, and in the case of a blocklist lookup that doesn't match on any blocklist entry, we need to not send any response at all.